### ISSUE TITLE
Decimal type fields cannot be resized or precision improved within the partition table

### ISSUE THREAD

**Issue by tonyhu214** [pingcap/tidb#54441]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
 create table test(
 id int not null primary key auto_increment,
 cost decimal(10,0) default 0 not null
 )engine = innodb
 partition by hash(id) 
 PARTITIONS  4 ;

alter table test modify column cost decimal(20,4) default 0 not null;

### 2. What did you expect to see? (Required)
sql error : Unsupported modify column: table is partition table

### 3. What did you see instead (Required)
Change fields to   decimal(20,4) 

### 4. What is your TiDB version? (Required)
6.5.0 +

**Comment by River2000i** [comment]
seems block from https://github.com/pingcap/tidb/blob/7d5f1d75bd350d5caa213a0aa8d6fec169553429/pkg/ddl/ddl_api.go#L6006-L6014

**Comment by tonyhu214** [comment]
Of course. MySQL supports changes to the decimal type of partitioned tables. I hope Tidb can also support it.

**Comment by lizhenhuan** [comment]
Will this issue be fixed in version 8.5?
I hope this issue can be resolved in version 8.5.
